agent/background.go

```diff
@@ -129,3 +131,22 @@ func (a *Agent) withCallbackTimeout(ctx context.Context, tc *taskContext) (conte
 	}
 	return context.WithTimeout(ctx, timeout)
 }
+
+func (a *Agent) startSpotTerminationWatcher(ctx context.Context, tc *taskContext, cancel context.CancelFunc) {
+	defer recovery.LogStackTraceAndContinue("spot termination watcher")
+	defer cancel()
+	ticker := time.NewTicker(10 * time.Second)
```

Comment:
what's the reason for waiting to exit?

Reply:
I put this in a ticker so that we don't bombard the API endpoint in a for loop. There's a 2-minute warning for termination, so polling every 10 seconds should give the agent plenty of time.